src/lib.rs

```diff
@@ -235,6 +235,28 @@ impl Surface {
         })
     }
 
+    /// Creates a new instance of this struct, using the provided [`HtmlCanvasElement`](web_sys::HtmlCanvasElement).
+    #[cfg(target_arch = "wasm32")]
+    pub fn from_canvas(canvas: web_sys::HtmlCanvasElement) -> Result<Self, SoftBufferError> {
+        let imple = SurfaceDispatch::Web(web::WebImpl::from_canvas(canvas)?);
+
+        Ok(Self {
+            surface_impl: Box::new(imple),
+        })
+    }
```

Comment:
The general pattern in `winit` is to make this an extension trait on the existing type. Could we do that here?